Add urls for change_password modal and login modal.

Withington opened the pull request:
Add the ability to get to the Change Password screen directly from
https://......../#change_password
useful for when the admin has given a user a new password and wants to direct the user to immediately change their password.

Also added url to go directly to login screen
https://......./#login

Needs improvement - 
This works fine if your browser tab is not already on Phenopolis. However, if your tab is already open at Phenopolis home screen, then you need to enter the url https://......../#change_password **and** hit refresh in order to get the modal to pop up.

Withington committed:
Add urls for login modal and change_password modal.

IsmailM commented:
Thanks for this 😄

With regards to your point, I think this is good enough tbh - especially since most users will only come across those links via a clickable URL in an email.

Alternatively, the proper fix would be to create another two endpoints - `/login` and `/change_password` with the method set to `['GET']` where you simply redirect the correct URL - i.e. `https://.../#login` or `https://.../#change_password `...

IsmailM commented on static/assets/js/home.js:
```diff
@@ -67,6 +67,13 @@ if (!PP) {
     });
 
     $(document).ready(function () {
```
Is it necessary to have a `$(document).ready()` here? It should work without?

The `PP.initLogin()` function is called from the html - see [here](https://github.com/phenopolis/phenopolis/blob/master/templates/material_layout.html#L89)...

Note - `$(document).ready()` will be executed once the document is loaded.  `$(function(){...});` (what we have in the html) is a shortcut for `$(document).ready()` and does the exact same thing. On the other hand, `(function(){...})();` will be executed as soon as it is encountered in the Javascript (i.e. the `PP` JS module) - basically invoking itself. 

AS such there shouldn't be any need have another `document.ready` function.

Also, if possible move the indexOf lines into a separate function and call it from the initLogin or directly from the [html](https://github.com/phenopolis/phenopolis/blob/master/templates/material_layout.html#L89). 

I typically prefer to just define functions in the JS library files and then have a single `on.ready` function in the HTML where I call the necessary functions...

Withington replied:
Thanks, I've made those changes.

Withington committed:
Added new url endpoints for change_password and login.

Withington commented:
Added https://......../change_password and https://......../login. These **do** refresh the browser when called.

Removed $(document).ready() and put the functionality in to initLogin() instead.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.

IsmailM commented on views/__init__.py:
```diff
@@ -180,6 +180,11 @@ def login():
 # 
+@app.route('/login', methods=['GET'])
+def login_form():
+    return redirect('/#login')
```
So we need an if statement here and below as the one here: https://github.com/phenopolis/phenopolis/blob/master/views/__init__.py#L187

IsmailM replied:
tbh not sure why that if statement is there but I suspect that it's there because redirecting to the home page caused some issues on the server.

IsmailM replied:
After that I'm happy with this PR 🙂

Withington replied:
Well spotted. Done, thanks.

Withington committed:
Added 'Search Phenopolis' button that becomes visible after a successful password change.
